src/main/include/controllers/DrivetrainController.hpp

```diff
@@ -354,12 +397,26 @@ class DrivetrainController : public ControllerBase<7, 2, 4> {
         frc::MakeCostMatrix(0.0625, 0.125, 2.5, 0.95, 0.95);
     Eigen::Matrix<double, 2, 2> m_R = frc::MakeCostMatrix(12.0, 12.0);
 
+    frc::LinearQuadraticRegulator<3, 2> m_turningLQR{
+        TurningDynamics(),
+        {0.001, 4.0, 4.0},
+        {12.0, 12.0},
+        Constants::kControllerPeriod};
+    frc::LinearPlantInversionFeedforward<3, 2> m_turningFF{
+        TurningDynamics(), Constants::kControllerPeriod};
+
+    frc::TrapezoidProfile<units::radian>::State m_headingState;
+    frc::TrapezoidProfile<units::radian>::Constraints m_constraints{kTMaxV,
+                                                                    kTMaxA};
+    frc::TrapezoidProfile<units::radian>::State m_profiledReference;
```

Comment:
I suggest putting these in a separate controller class for separation of responsibilities.